commands/cloudapi-v5/ipblock.go

```diff
@@ -232,22 +240,72 @@ func RunIpBlockUpdate(c *core.CommandConfig) error {
 }
 
 func RunIpBlockDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete ipblock"); err != nil {
-		return err
-	}
-	c.Printer.Verbose("Ip block with id: %v is deleting...", viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgIpBlockId)))
-	resp, err := c.CloudApiV5Services.IpBlocks().Delete(viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgIpBlockId)))
-	if resp != nil {
-		c.Printer.Verbose(config.RequestTimeMessage, resp.RequestTime)
+	var resp *resources.Response
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv5.ArgAll))
+	if allFlag {
+		err := DeleteAllIpBlocks(c)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete ipblock"); err != nil {
+			return err
+		}
+		c.Printer.Verbose("Ip block with id: %v is deleting...", viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgIpBlockId)))
+		resp, err := c.CloudApiV5Services.IpBlocks().Delete(viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgIpBlockId)))
+		if resp != nil {
+			c.Printer.Verbose(config.RequestTimeMessage, resp.RequestTime)
+		}
+		if err != nil {
+			return err
+		}
+
+		if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
+			return err
+		}
 	}
+	return c.Printer.Print(getIpBlockPrint(resp, c, nil))
+}
+
+func DeleteAllIpBlocks(c *core.CommandConfig) error {
+	_ = c.Printer.Print("IpBlocks to be deleted:")
+	ipBlocks, resp, err := c.CloudApiV5Services.IpBlocks().List()
 	if err != nil {
 		return err
 	}
+	if ipBlocksItems, ok := ipBlocks.GetItemsOk(); ok && ipBlocksItems != nil {
+		for _, dc := range *ipBlocksItems {
+			if id, ok := dc.GetIdOk(); ok && id != nil {
+				_ = c.Printer.Print("IpBlock Id: " + *id)
+			}
+			if properties, ok := dc.GetPropertiesOk(); ok && properties != nil {
+				if name, ok := properties.GetNameOk(); ok && name != nil {
+					_ = c.Printer.Print(" IpBlock Name: " + *name)
+				}
+			}
+		}
 
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete all the IpBlocks"); err != nil {
+			return err
+		}
+		c.Printer.Verbose("Deleting all the IpBlocks...")
+		for _, dc := range *ipBlocksItems {
+			if id, ok := dc.GetIdOk(); ok && id != nil {
+				c.Printer.Verbose("Deleting IpBlock with id: %v...", *id)
+				resp, err = c.CloudApiV5Services.IpBlocks().Delete(*id)
+				if resp != nil {
+					c.Printer.Verbose(config.RequestTimeMessage, resp.RequestTime)
+				}
+				if err != nil {
+					return err
+				}
+				if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
+					return err
+				}
+			}
+		}
 	}
-	return c.Printer.Print(getIpBlockPrint(resp, c, nil))
```

Comment:
Deleting this line, leaves the user without a feedback if the command has been executed and it looks something like this:
```
ionosctl ipblock delete -i 33f5029a-e802-4af4-95c1-a47df0445ab6 
Warning: Are you sure you want to delete ipblock (y/N) ? 
y
```
From the current behaviour: 
```
ionosctl ipblock delete -i 8caca11d-fb68-4160-843b-cf485f77fa6d 
Warning: Are you sure you want to delete ipblock (y/N) ? 
y
RequestId: b390b67b-a5c5-47a5-8709-979109bdba45
Status: Command ipblock delete has been successfully executed
```
In my opinion, it is better that the user has some feedback regarding a command, for both delete options (--all and --resource-id). What is your opinion on this? 

(This applies to ALL the places in the PR where this line has been deleted)

Reply:
I think is a git misunderstanding, because I do not removed this line, is still there, and every resource has it,s return line that uses the Printer method

Reply:
In my opinion, for the feedback to be given for both delete all and for delete simple - the `resp` variable needs to be returned as well on `DeleteAll<Resource>` function, and also to not be overwritten/defined as a new variable in the delete "simple" functionality.